packages/shared-components/src/SelectFieldArray.tsx

```diff
@@ -0,0 +1,126 @@
+  fields: FieldArrayFieldsProps<EmptySelectTemplate>,
+  emptySelectTemplate: EmptySelectTemplate,
+) => () => {
+  fields.push(emptySelectTemplate);
+};
+
+const onKeyDown = (fields: FieldArrayFieldsProps<EmptySelectTemplate>, emptySelectTemplate: EmptySelectTemplate) => ({
+  keyCode,
+}) => {
+  if (keyCode === 13) {
+    fields.push(emptySelectTemplate);
+  }
+};
+
+const getRemoveButton = (index: number, fields: FieldArrayFieldsProps<EmptySelectTemplate>) => className => {
+  if (index > 0) {
+    return (
+      <button
+        className={className || styles.buttonRemove}
+        type="button"
+        onClick={() => {
+          fields.remove(index);
+        }}
+      />
+    );
+  }
+  return undefined;
+};
+
+const showErrorMessage = meta => meta && meta.error && (meta.dirty || meta.submitFailed);
+
+/**
+ * SelectFieldArray
+ *
+ * Overbygg over FieldArray (Redux-form) som håndterer å legge til og fjerne perioder
+ */
+const SelectFieldArray = ({
+  intl,
+  fields,
+  readOnly,
+  meta,
+  titleTextCode,
+  textCode,
+  emptySelectTemplate,
+  shouldShowAddButton,
+  createAddButtonInsteadOfImageLink,
+  children,
+}: SelectFieldArrayProps & WrappedComponentProps) => (
+  <NavFieldGroup
+    title={titleTextCode ? intl.formatMessage({ id: titleTextCode }) : undefined}
+    errorMessage={showErrorMessage(meta) ? intl.formatMessage(...meta.error) : null}
+  >
+    {fields.map((selectElementFieldId, index) => children(selectElementFieldId, index, getRemoveButton(index, fields)))}
+    {shouldShowAddButton && (
+      <Row className="">
+        <Column xs="12">
+          {!createAddButtonInsteadOfImageLink && !readOnly && (
+            // eslint-disable-next-line jsx-a11y/click-events-have-key-events
+            <div
+              onClick={onClick(fields, emptySelectTemplate)}
+              onKeyDown={onKeyDown(fields, emptySelectTemplate)}
+              className={styles.addSelect}
+              role="button"
+              tabIndex={0}
+            >
+              <Image className={styles.addCircleIcon} src={addCircleIcon} alt={intl.formatMessage({ id: textCode })} />
+              <Undertekst className={styles.imageText}>
+                <FormattedMessage id={textCode} />
+              </Undertekst>
+            </div>
+          )}
+          {createAddButtonInsteadOfImageLink && !readOnly && (
+            <button type="button" onClick={onClick(fields, emptySelectTemplate)} className={styles.buttonAdd}>
+              <FormattedMessage id={textCode} />
+            </button>
+          )}
+          <VerticalSpacer sixteenPx />
+        </Column>
+      </Row>
+    )}
+  </NavFieldGroup>
+);
+
+SelectFieldArray.defaultProps = {
+  readOnly: true,
+  titleTextCode: undefined,
+  meta: undefined,
+  textCode: 'SelectFieldArray.LeggTilSelect',
+  emptySelectTemplate: { value: '' },
+  shouldShowAddButton: true,
+  createAddButtonInsteadOfImageLink: false,
+};
+
+export default injectIntl(SelectFieldArray);
```

Comment:
useIntl hook fra 'react-intl' kan brukes for å slippe å wrappe komponenten og eventuell prop drilling

Reply:
smart, bytter til den